### ISSUE TITLE
Controller actions ending with words like "action" are removing the "action" part from route name

### ISSUE THREAD

**Issue by moynzzz** [symfony/symfony#40024]
**Symfony version(s) affected**: All

**Description**  
If you have in a controller some action like this one:

```php
/**
 * @Route("/create-transaction")
 */
public function createTransaction()
{
    // Code here
}
```

Then if you have to generate a route the action name will be `app_controller_createtrans`. If you want to have a proper action name like `app_controller_createtransaction` you should have the name of the action to `createTransactionAction`.

I know this is not exactly a bug and we can introduce a BC. It will be better for newcomers if the automatic name is the same as the method name except if it ends with an `Action` with capital `A`. WDYT? If someone decides that this should be done I want to propose a PR.

**Possible Solution**

**Comment by OskarStark** [comment]
It looks like we are checking for "action" case insensitive, maybe it's better to just check for "Action" here?

**Comment by nicolas-grekas** [comment]
I agree this should be case sensitive. PR welcome.

**Comment by moynzzz** [comment]
Nice I'm working on the PR.

**Comment by moynzzz** [comment]
I find this harder than expected. The search for lowercased "action" string is in `AnnotatedRouteControllerLoader` but it is using the string provided from `AnnotationClassLoader`'s `getDefaultRouteName` function which is always lowercased. We can't actually search for "Action" with capital 'a' when we always have lowercased route name.

The only idea that I have is to override the function `getDefaultRouteName` entirely in `AnnotatedRouteControllerLoader` but this will copy the same code. What do you think?

**Comment by nicolas-grekas** [comment]
Thanks for having a look and for the report. I'm proposing a fix in #41782
